lib/IRGen/LoadableByAddress.cpp

```diff
@@ -1491,7 +1491,19 @@ void LoadableStorageAllocation::allocateForArg(SILValue value) {
 
   assert(!ApplySite::isa(value) && "Unexpected instruction");
 
-  SILBuilderWithScope allocBuilder(&*pass.F->begin());
+  // Find the first non-alloc stack and use its scope when creating
+  // the new SILBuilder. AllocStackInst(s) can be hoisted and mess
+  // up with scopes.
```

Comment:
That's true, but the real reason is:
An alloc_stack does not directly cause any code to be generated. The location of an alloc_stack carries information about the source variable; it doesn't matter where in the instruction stream the alloc_stack is located.